src/app/content/components/Toolbar/PracticeQuestions.spec.tsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import { Provider } from 'react-redux';
+import renderer from 'react-test-renderer';
+import createTestServices from '../../../../test/createTestServices';
+import createTestStore from '../../../../test/createTestStore';
+import { receiveFeatureFlags } from '../../../actions';
+import * as Services from '../../../context/Services';
+import MessageProvider from '../../../MessageProvider';
+import { Store } from '../../../types';
+import { practiceQuestionsFeatureFlag } from '../../constants';
+import PracticeQuestionsButton, { PracticeQuestionsWrapper } from './PracticeQuestionsButton';
+
+describe('practice questions button', () => {
+  let store: Store;
+  let services: ReturnType<typeof createTestServices>;
+
+  beforeEach(() => {
+    store = createTestStore();
+    services = createTestServices();
+  });
+
+  it('does not render if feature flag is not enabled', () => {
+    store.dispatch(receiveFeatureFlags([]));
+
+    const component = renderer.create(<Provider store={store}>
+      <Services.Provider value={services}>
+        <MessageProvider>
+          <PracticeQuestionsButton />
+        </MessageProvider>
+      </Services.Provider>
+    </Provider>);
+
+    expect(component.toJSON()).toMatchSnapshot();
+  });
+
+  it('render if feature flag is enabled', () => {
+    store.dispatch(receiveFeatureFlags([practiceQuestionsFeatureFlag]));
+
+    const component = renderer.create(<Provider store={store}>
+      <Services.Provider value={services}>
+        <MessageProvider>
+          <PracticeQuestionsButton />
+        </MessageProvider>
+      </Services.Provider>
+    </Provider>);
+
+    expect(component.toJSON()).toMatchSnapshot();
+  });
+
+  it('clicking button works', () => {
+    store.dispatch(receiveFeatureFlags([practiceQuestionsFeatureFlag]));
+
+    const component = renderer.create(<Provider store={store}>
```

Comment:
what about putting all renderer.create into a render() function?

Reply:
I think this is a good idea if all of the tests use the same JSX